pkg/ipamd/datastore/data_store.go

```diff
@@ -546,9 +645,23 @@ func (ds *DataStore) unassignPodIPv4AddressUnsafe(addr *AddressInfo) {
 	assignedIPs.Set(float64(ds.assigned))
 }
 
-// GetStats returns total number of IP addresses and number of assigned IP addresses
-func (ds *DataStore) GetStats() (int, int) {
-	return ds.total, ds.assigned
+// GetStats returns total number of IP addresses, number of assigned IP addresses and total prefixes
+func (ds *DataStore) GetStats() (int, int, int) {
+	ds.lock.Lock()
+	defer ds.lock.Unlock()
+
+	totalIPs := 0
+	assignedIPs := 0
+	for _, eni := range ds.eniPool {
+		for _, cidr := range eni.AvailableIPv4Cidrs {
+			if (ds.isPDEnabled && cidr.IsPrefix) || (!ds.isPDEnabled && !cidr.IsPrefix) {
```

Comment:
nit: this `(ds.isPDEnabled && cidr.IsPrefix) || (!ds.isPDEnabled && !cidr.IsPrefix)` happens multiple times.
It can be replaced with something like `IsCIDReligibleForAllocation`

Reply:
As discussed, will take this up post GA.